src/pal/src/exception/machexception.cpp

```diff
@@ -189,142 +245,58 @@ PAL_ERROR CorUnix::CPalThread::EnableMachExceptions()
         // some or all of the handlers might still be ours. In those cases we don't want to overwrite the
         // chain-back entries with these useless self-references.
-        kern_return_t MachRet;
-        mach_msg_type_number_t oldCount = CThreadMachExceptionHandlerNode::s_nPortsMax;
-        exception_mask_t rgMasks[CThreadMachExceptionHandlerNode::s_nPortsMax];
-        exception_handler_t rgHandlers[CThreadMachExceptionHandlerNode::s_nPortsMax];
-        exception_behavior_t rgBehaviors[CThreadMachExceptionHandlerNode::s_nPortsMax];
-        thread_state_flavor_t rgFlavors[CThreadMachExceptionHandlerNode::s_nPortsMax];
+        kern_return_t machret;
+        kern_return_t machretDeallocate;
         thread_port_t thread = mach_thread_self();
-        exception_behavior_t excepBehavior = EXCEPTION_STATE_IDENTITY;
-
-        MachRet = thread_swap_exception_ports(thread,
-                                              machExceptionMask,
-                                              s_ExceptionPort,
-                                              excepBehavior,
-                                              MACHINE_THREAD_STATE,
-                                              rgMasks,
-                                              &oldCount,
-                                              rgHandlers,
-                                              rgBehaviors,
-                                              rgFlavors);
-
-        kern_return_t MachRetDeallocate = mach_port_deallocate(mach_task_self(), thread);
-        CHECK_MACH("mach_port_deallocate", MachRetDeallocate);
 
-        if (MachRet != KERN_SUCCESS)
+        machret = thread_swap_exception_ports(
+            thread,
+            machExceptionMask,
+            s_ExceptionPort,
+            EXCEPTION_DEFAULT | MACH_EXCEPTION_CODES,
+            THREAD_STATE_NONE,
+            pSavedHandlers->m_masks,
+            &pSavedHandlers->m_nPorts,
+            pSavedHandlers->m_handlers,
+            pSavedHandlers->m_behaviors,
+            pSavedHandlers->m_flavors);
+
+        machretDeallocate = mach_port_deallocate(mach_task_self(), thread);
+        CHECK_MACH("mach_port_deallocate", machretDeallocate);
+
+        if (machret != KERN_SUCCESS)
         {
-            ASSERT("thread_swap_exception_ports failed: %d\n", MachRet);
-            return UTIL_MachErrorToPalError(MachRet);
+            ASSERT("thread_swap_exception_ports failed: %d %s\n", machret, mach_error_string(machret));
+            return UTIL_MachErrorToPalError(machret);
         }
 
-        // Scan through the returned handlers looking for those that are ours.
-        for (mach_msg_type_number_t i = 0; i < oldCount; i++)
+#ifdef _DEBUG
+        NONPAL_TRACE("EnableMachExceptions: THREAD PORT count %d\n", pSavedHandlers->m_nPorts);
+        for (mach_msg_type_number_t i = 0; i < pSavedHandlers->m_nPorts; i++)
         {
-            if (rgHandlers[i] == s_ExceptionPort)
```

Comment:
I am not sure I understand why we don't need to have this deleted code anymore. Could you please explain it?